modules/apps/map/map-common/test/liferay/MapBase.es.js

```diff
@@ -21,6 +21,14 @@ describe('MapBase', () => {
 	let bounds;
 
 	class MapImpl extends MapBase {
+		set position(position) {
+			this._STATE_.position = position;
+		}
+
+		get position() {
+			return getLocation();
+		}
+
```

Comment:
This is a bit of a smell here. Why do you need to override the get/set from the MapBase class you are extending from?

Reply:
@bryceosterhaus without these, the test complains:
```
    TypeError: Cannot read properties of undefined (reading 'address')

      69 |              this.emit('positionChange', {
      70 |                      newVal: {
    > 71 |                              address: position.address,
         |                                                ^
      72 |                              location: position.location,
      73 |                      },
      74 |              });
```

I'll try to remove the getter and setter and find an alternative, but I couldn't the last time I tried.

Reply:
> without these, the test complains:

Yeah, that is probably indicative that something is off with the refactor. When in the world of migrating and refactoring code, always treat the initial passing unit tests as the source of truth. You shouldn't have to change anything in existing tests during a refactor. The reason for this is because the tests are simply a consumer of an API, meaning that other people and customers may be consuming it in the same way. So if tests are failing and require changes, that means a customers code would also fail and require changes, which is not something we want.

Reply:
Yeah, that makes sense and I didn't make this change lightly. I was "blaming" or "justifying" it because of a possible infrastructural change from metal's state/prop management to the EventEmitter/vanilla JS.

I'll figure it out by playing around with the implementation.